lib/search/index.ts

```diff
@@ -445,8 +447,14 @@ export const middleware: S.Middleware = (store) => {
         const oldHash = t(action.oldTagName);
         const newHash = t(action.newTagName);
 
-        if (searchState.openedTag === oldHash) {
-          searchState.openedTag = newHash;
+        if (
+          searchState.collection.type === 'tag' &&
+          searchState.collection.tagName === action.oldTagName
+        ) {
+          searchState.collection = {
+            type: 'tag',
+            tagName: action.newTagName,
```

Comment:
very nice catch!

though here in the search middleware it should be fine to rely on the `tagHash`, since we're not _displaying_ the tag. everything compared against for searching and filtering will rely on the `tagHash`

Reply:
Good point; I wanted to keep the signature the same in both though in case maybe later all the searchState could be placed in redux.